clang/lib/Driver/Driver.cpp

```diff
@@ -5803,12 +5808,12 @@ class OffloadingActionBuilder final {
           // post link is not optional - even if not splitting, always need to
           // process specialization constants
           types::ID PostLinkOutType = isSPIR || isSYCLNativeCPU
-                                          ? types::TY_Tempfiletable
+                                          ? types::TY_Filetable
                                           : types::TY_LLVM_BC;
           auto createPostLinkAction = [&]() {
-            // For SPIR-V targets, force TY_Tempfiletable.
+            // For SPIR-V targets, force TY_Filetable.
             auto TypedPostLinkAction = C.MakeAction<SYCLPostLinkJobAction>(
-                FullDeviceLinkAction, PostLinkOutType, types::TY_Tempfiletable);
+                FullDeviceLinkAction, PostLinkOutType, types::TY_Filetable);
```

Comment:
This area should remain `TY_Tempfiletable` as you are working with items after the device link and during `sycl-post-link` which are dealing with LLVM-IR, not SPIR-V.